source/extensions/http/header_validators/envoy_default/header_validator.cc

```diff
@@ -8,38 +14,344 @@ namespace EnvoyDefault {
 
 using ::envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig;
 using ::Envoy::Http::HeaderString;
+using ::Envoy::Http::Protocol;
+
+HeaderValidator::HeaderValidator(const HeaderValidatorConfig& config, Protocol protocol,
+                                 StreamInfo::StreamInfo& stream_info)
+    : config_(config), protocol_(protocol), stream_info_(stream_info),
+      header_values_(::Envoy::Http::Headers::get()) {}
+
+::Envoy::Http::HeaderValidator::HeaderEntryValidationResult
+HeaderValidator::validateMethodHeader(const HeaderString& value) {
+  // HTTP Method Registry, from iana.org:
+  // source: https://www.iana.org/assignments/http-methods/http-methods.xhtml
+  //
+  // From the RFC:
+  //
+  // tchar = "!" / "#" / "$" / "%" / "&" / "'" / "*" / "+" / "-" / "."
+  //       /  "^" / "_" / "`" / "|" / "~" / DIGIT / ALPHA
+  // token = 1*tchar
+  // method = token
+  //
+  static absl::node_hash_set<absl::string_view> kHttpMethodRegistry = {
```

Comment:
So @yanavlasov if we think we're moving this to quiche in the long run (which I think we want to?  cc @ianswett ) I wonder if we want a node hash map and include if they're safe / idempotent as well.  cc @danzh2010 for thoughts as well

Reply:
yes, I am fairly sure we want to move this to quiche eventually.